### PR TITLE
Alex Schwartz

### PR DESCRIPTION
**Pull request by aschwartz1**

### CODE REVIEW

File: lib/character.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Character
+  def initialize(information)
+    @information = information
```
Comment (mikedao) [inline lib/character.rb:3]: In general you want to not save the hash but split it up into individual instance variables so you can leverage the attr_reader helper methods

File: lib/show.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def total_salary
+    @characters.sum do |character|
+      character.salary
+    end
+  end
+
+  def highest_paid_actor
+    @characters.max(&:salary).actor
```
Comment (mikedao) [inline lib/show.rb:21]: This doesn't work, you want max_by, not max. You should also use the multi-line syntax here.

**Review by mikedao — COMMENTED** [review]
Overall good.  I would avoid using mocks and stubs until they are necessary. That's why your highest_paid_actor test is passing when it should be failing.

File: lib/character.rb (hunk 1 above)
Comment (aschwartz1) [reply lib/character.rb:3]: @mikedao doesn't that go back to the same design choice you and Sal were discussing in the previous IC review -- do you lose the original hash or do you keep it?  As in, it's 6 this way, half-a-dozen the other?

File: lib/show.rb (hunk 2 above)
Comment (aschwartz1) [reply lib/show.rb:21]: Understood.

**Comment by aschwartz1** [comment]
@mikedao re: using mocks

I tried to make my tests not rely on the other classes, thinking that was dandy OOP and a challenge from the lesson on Monday.  Definitely feels like that backfired.

Talked to my mentor, perhaps now I know a better use case for mocks...

**Pull request closed without merging**